File: v2/sdk/javascript/examples/basic.ts
/**
 * Basic prediction example.
 * Run: npx tsx examples/basic.ts
 */
import { SimulatorClient, SimulatorError } from '../index';

async function main() {
  const client = new SimulatorClient({
    baseUrl: 'https://simulator.vercel.app',
  });

  try {
    const result = await client.predict({
      scenario: 'Launch a mobile app for language learning targeting Southeast Asia',
    });

    console.log('--- Prediction Result ---');
    console.log(`Scenario: ${result.scenario}`);
    console.log(`Probability: ${result.probability}%`);
    console.log(`Confidence: ${result.confidence}`);
    console.log(`Range: ${result.probRange.adverse}% (adverse) - ${result.probRange.optimistic}% (optimistic)`);
    console.log('');
    console.log('Key Bottlenecks:');
    for (const b of result.keyBottlenecks) {
      console.log(`  [${b.prob}%] ${b.label} -- ${b.description}`);
    }
    console.log('');
    console.log('Sources:', result.sources.join(', '));
    console.log('Sacred Roots:', result.sacredRoots.join(', '));
    console.log('Generated at:', result.generatedAt);
  } catch (err) {
    if (err instanceof SimulatorError) {
      console.error(`API Error (${err.statusCode}): ${err.message}`);
    } else {
      console.error('Unexpected error:', err);
    }
  }
}

main();

File: v2/sdk/python/simulator_sdk/examples/basic.py
"""
Basic prediction example.
Run: python -m simulator_sdk.examples.basic
"""
from simulator_sdk import SimulatorClient, SimulatorError


def main():
    client = SimulatorClient(base_url="https://simulator.vercel.app")

    try:
        result = client.predict(
            scenario="Launch a mobile app for language learning targeting Southeast Asia",
        )

        print("--- Prediction Result ---")
        print(f"Scenario: {result.scenario}")
        print(f"Probability: {result.probability}%")
        print(f"Confidence: {result.confidence}")
        print(
            f"Range: {result.prob_range['adverse']}% (adverse) "
            f"- {result.prob_range['optimistic']}% (optimistic)"
        )
        print()
        print("Key Bottlenecks:")
        for b in result.key_bottlenecks:
            print(f"  [{b.prob}%] {b.label} -- {b.description}")
        print()
        print("Sources:", ", ".join(result.sources))
        print("Sacred Roots:", ", ".join(result.sacred_roots))
        print("Generated at:", result.generated_at)

    except SimulatorError as e:
        print(f"API Error ({e.status_code}): {e}")


if __name__ == "__main__":
    main()
